src/lib.rs

```diff
@@ -174,10 +201,24 @@ impl Picture {
         (*self.pic).stride[component] as i32
     }
 
-    pub fn plane_data(&self, component: usize) -> *mut c_void {
+    pub fn plane_data_ptr(&self, component: usize) -> *mut c_void {
         (*self.pic).data[component]
     }
 
+    pub fn plane_data(&self, component: PlanarImageComponent) -> (&[u8], i32) {
+        let stride = match component {
+            PlanarImageComponent::Y => self.stride(0),
+            _ => self.stride(1)
+        };
+        let index: usize = component.into();
+        unsafe {
+            (std::slice::from_raw_parts(
+                (*self.pic).data[index] as *const u8,
+                (stride as i32 * self.height() as i32) as usize,
```

Comment:
This seems wrong. For some formats the height of the chroma planes will be less than the frame height, e.g. for 4:2:0 subsampling.

Reply:
Also why first casting to `i32` just to cast the overall result to `usize`? Wouldn't it make more sense to cast to `usize` directly?